### PR TITLE
[Integration Tests] Added second interceptor test for node disconnects

### PR DESCRIPTION
**Pull request by fassadlr**
This covers the scenario where minerA/B/C are synced to height 5. Miner C mines on its to height 10. When miner's A an B want to reorg to miner C's chain and their chain data has rewound to block 5, we intercept it and disconnect miner C. Miner's A and B should still synced to block 10.

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs [hunk 1]
```diff
@@ -67,9 +70,67 @@ bool interceptor(ChainedHeaderBlock chainedHeaderBlock)
...
+                    }
+
+                    return false;
+                }
+
+                minerA.BlockDisconnectInterceptor(interceptor);
+
+                // Start the nodes.
+                minerA.Start();
+                minerB.Start();
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs]: miner B never mines perhaps better to keep the terminology of MinerA, MinerB and Syncer

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs [hunk 2]
```diff
@@ -67,9 +70,67 @@ bool interceptor(ChainedHeaderBlock chainedHeaderBlock)
...
+                minerC.Start();
+
+                // MinerA mines 5 blocks.
+                TestHelper.MineBlocks(minerA, 5);
+
+                // MinerB/C syncs with MinerA.
+                TestHelper.ConnectAndSync(minerA, minerB);
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs]: Use the overload ll nodes in one line

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs [hunk 3]
```diff
@@ -67,9 +70,67 @@ bool interceptor(ChainedHeaderBlock chainedHeaderBlock)
...
+                TestHelper.MineBlocks(minerA, 4);
+                TestHelper.WaitLoop(() => minerA.FullNode.ConsensusManager().Tip.Height == 9);
+                TestHelper.WaitLoop(() => minerB.FullNode.ConsensusManager().Tip.Height == 9);
+                TestHelper.WaitLoop(() => minerC.FullNode.ConsensusManager().Tip.Height == 5);
+
+                // MinerC mines 5 more blocks so that a reorg is triggered.
+                TestHelper.MineBlocks(minerC, 5);
+
+                // Miner A and B should have reorged to the longer chain.
+                TestHelper.WaitLoop(() => TestHelper.IsNodeSyncedAtHeight(minerA, 10));
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs]: Here is better to check A,B are synced with C then against a specific height.

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs]: Ah yes, good spot. Will rename 👍

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs (hunk 2 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs]: 👍

File: src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs (hunk 3 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/ConsensusManagerInterceptorTests.cs]: Done!

**Pull request merged**